pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -83,6 +111,49 @@ func (pg *pgcontroller) updateReplicaSet(oldObj, newObj interface{}) {
 	pg.addReplicaSet(newObj)
 }
 
+func (pg *pgcontroller) addStatefulSet(obj interface{}) {
+	sts, ok := obj.(*appsv1.StatefulSet)
+	if !ok {
+		klog.Errorf("Failed to convert %v to appsv1.StatefulSet", obj)
+		return
+	}
+
+	if *sts.Spec.Replicas == 0 {
+		pgName := batchv1alpha1.PodgroupNamePrefix + string(sts.UID)
+		err := pg.vcClient.SchedulingV1beta1().PodGroups(sts.Namespace).Delete(context.TODO(), pgName, metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			klog.Errorf("Failed to delete PodGroup <%s/%s>: %v", sts.Namespace, pgName, err)
+		}
+	}
+}
+
+func (pg *pgcontroller) updateStatefulSet(oldObj, newObj interface{}) {
+	pg.addStatefulSet(newObj)
+}
+
+func (pg *pgcontroller) addJob(obj interface{}) {
+	job, ok := obj.(*batchv1.Job)
+	if !ok {
+		klog.Errorf("Failed to convert %v to batchv1.Job", obj)
+		return
+	}
+	// Delete PodGroup if the job is Complete or Failed or Suspended.
+	for _, condition := range job.Status.Conditions {
+		if condition.Status == v1.ConditionTrue && (condition.Type == batchv1.JobComplete || condition.Type == batchv1.JobFailed || condition.Type == batchv1.JobSuspended) {
```

Comment:
Better abstract a function here like jobTerminated to increase code readability, and so does pod.